rust/azure_iot_operations_connector/examples/base_connector_sample.rs

```diff
@@ -56,86 +57,86 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
-                    asset_client.asset_ref().name
-                );
-                Err(AdrConfigError {
-                    message: Some("asset manufacturer type is not supported".to_string()),
-                    ..AdrConfigError::default()
-                })
-            }
-        };
-
-        asset_client.report_status(asset_status).await;
+async fn run_device(
+    mut device_endpoint_client: DeviceEndpointClient,
+    mut asset_creation_observation: AssetClientCreationObservation,
+) {
+    loop {
+        tokio::select! {
+            biased;
+            // Listen for a device update notifications
+            res = device_endpoint_client.recv_update() => {
+                if let Some(()) = res {
+                    log::info!("Device updated: {device_endpoint_client:?}");
+                    // now we should update the status of the device
+                    let endpoint_status = generate_endpoint_status(&device_endpoint_client);
 
-        for dataset in asset_client.datasets() {
-            tokio::task::spawn({
-                let dataset_clone = dataset.clone();
-                async move {
-                    run_dataset(dataset_clone).await;
+                    device_endpoint_client
+                        .report_status(Ok(()), endpoint_status)
+                        .await;
+                } else {
+                    log::error!("No more Device Endpoint updates will be received");
+                    break;
+                }
+            }
+            // Listen for a asset creation notifications
+            res = asset_creation_observation.recv_notification() => {
+                if let Some((asset_client, _asset_update_observation, _asset_deletion_token)) = res {
+                    log::info!("Asset created: {asset_client:?}");
+
+                // now we should update the status of the asset
+                let asset_status = match asset_client.specification().manufacturer.as_deref() {
+                    Some("Contoso") | None => Ok(()),
+                    Some(m) => {
+                        log::warn!(
+                            "Asset '{}' not accepted. Manufacturer '{m}' not supported.",
+                            asset_client.asset_ref().name
+                        );
+                        Err(AdrConfigError {
+                            message: Some("asset manufacturer type is not supported".to_string()),
+                            ..AdrConfigError::default()
+                        })
+                    }
+                };
+
+                asset_client.report_status(asset_status).await;
+
+                for dataset in asset_client.datasets() {
+                    tokio::task::spawn({
+                        let dataset_clone = dataset.clone();
+                        async move {
+                            run_dataset(dataset_clone).await;
+                        }
+                    });
+                }
+                } else {
```

Comment:
nit: same issue as the REST PR, formatting is getting wonky in the `select!`.

Reply:
gross, fixed. Thanks for catching!